tests/opensearch/basic.bats

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bats
+
+load "$HELPERS_ROOT/test-helpers.bash"
+
+ensure_variables_set BATS_SUPPORT_ROOT BATS_ASSERT_ROOT BATS_DETIK_ROOT BATS_FILE_ROOT TEST_NAMESPACE FLUENTBIT_IMAGE_TAG OPENSEARCH_IMAGE_TAG
+
+load "$BATS_DETIK_ROOT/utils.bash"
+load "$BATS_DETIK_ROOT/linter.bash"
+load "$BATS_DETIK_ROOT/detik.bash"
+load "$BATS_SUPPORT_ROOT/load.bash"
+load "$BATS_ASSERT_ROOT/load.bash"
+load "$BATS_FILE_ROOT/load.bash"
+
+setup() {
+    echo "recreating namespace $TEST_NAMESPACE"
+    run kubectl delete namespace "$TEST_NAMESPACE"
+    run kubectl create namespace "$TEST_NAMESPACE"
+}
+
+# These are required for bats-detik
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAME="kubectl -n $TEST_NAMESPACE"
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAMESPACE="${TEST_NAMESPACE}"
+
+
+@test "test fluent-bit forwards logs to opensearch default index" {
+    helm repo add opensearch https://opensearch-project.github.io/helm-charts/
+    helm repo add fluent https://fluent.github.io/helm-charts
+    helm repo update
+
+    helm upgrade --install --debug --namespace "$TEST_NAMESPACE" opensearch opensearch/opensearch -f ${BATS_TEST_DIRNAME}/resources/helm/opensearch-basic.yaml --set image.tag=${OPENSEARCH_IMAGE_TAG} --wait
+
+    run try "at most 15 times every 2s " \
+        "to find 1 pods named 'opensearch-cluster-master-0' " \
+        "with 'status' being 'running'"
+
+    assert_success
+
+    helm upgrade --install --debug --namespace "$TEST_NAMESPACE" fluent-bit fluent/fluent-bit -f ${BATS_TEST_DIRNAME}/resources/helm/fluentbit-basic.yaml --set image.tag=${FLUENTBIT_IMAGE_TAG} --wait
+
+    run try "at most 15 times every 2s " \
+        "to find 1 pods named 'fluent-bit' " \
+        "with 'status' being 'running'"
+
+    assert_success
+
+    attempt=0
+    while true; do
+    	run kubectl exec -q -n $TEST_NAMESPACE opensearch-cluster-master-0 -- curl --insecure -s -w "%{http_code}" https://admin:admin@localhost:9200/fluentbit/_search/ -o /dev/null
+        if [[ "$output" != "200" ]]; then
```

Comment:
This feels like a helper method: you want to check for a 200 code on a URL with a max attempt and delay between each. I think I had one...

Reply:
Yeah see `wait_for_url`, doesn't that just do all this?

Reply:
Oh right, you're kubectl exec'ing in: why don't we just expose a service to simplify?

Reply:
yeah, I'll add a service and use wait_for_url

Reply:
Not sure this is resolved...

Reply:
@patrick-stephens I don't know if we gain much from that helper, I could port-forward that port, but there is already a service existing on the opensearch deployment.

Reply:
I think you need to document what and why then. This looks like a smell at the moment.

It also requires an image with the commands you want to run in it, which may change.